### ISSUE TITLE
can we use create and unzip password protected ZIP ?

### ISSUE THREAD

**Issue by mianharisali** [zeroturnaround/zt-zip#93]

**Comment by nadouani** [comment]
Same question: does this library allow creating archives with password protection?

Thanks

**Comment by toomasr** [comment]
Wow, an unanswered issue. Yeah, no password support at the moment. Would need to do some research on this and see how doable is this.

**Comment by nadouani** [comment]
Cool thanks, we found just one library that supports that feature: zip4j but it seems to not be maintained :(

**Comment by reinra** [comment]
You could also call an external tool. [zt-exec](https://github.com/zeroturnaround/zt-exec) might be useful for that.

**Comment by toomasr** [comment]
Ok, cool! Btw, which feature are you looking for exactly. I see that https://en.wikipedia.org/wiki/Zip_(file_format)#Encryption has some options. Wondering which to look at for possible implementation.

**Comment by nadouani** [comment]
Any option is enough for our project (https://github.com/CERT-BDF/TheHive). We just want to generate Zip archives that require a password (independently from the encryption strength)

**Comment by nadouani** [comment]
> You could also call an external tool. zt-exec might be useful for that.

No, using an external tool is not a viable option for us unfortunately :(

**Comment by toomasr** [comment]
I looked into this a little. Right now I see that to support passwords and encryption we would need to implement the ZIP layer fully rather than relying on the JDK ZIP support that just doesn't support passwords. This sounds like a huge undertaking though and maybe even a separate project to this. And then this project can choose which subsystem to use for example.

Any thoughts?
